src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/DefaultEditorTemplates.cs

```diff
@@ -378,6 +378,54 @@ public static IHtmlContent MonthInputTemplate(IHtmlHelper htmlHelper)
             return GenerateTextBox(htmlHelper, inputType: "month");
         }
 
+        public static IHtmlContent WeekInputTemplate(IHtmlHelper htmlHelper)
+        {
+            // A new HTML5 input type that only will be rendered in Rfc3339 mode    
+            var value = htmlHelper.ViewData.Model;
+            var metadata = htmlHelper.ViewData.ModelMetadata;
+
+            if (htmlHelper.ViewData.TemplateInfo.FormattedModelValue == value || !metadata.HasNonDefaultEditFormat)
+            {
+                if (value is DateTime || value is DateTimeOffset)
+                {
+                    var date = DateTimeOffset.Parse(Convert.ToString(value)).DateTime;
+                    var calendar = System.Threading.Thread.CurrentThread.CurrentCulture.Calendar;
+
+                    var year = calendar.GetYear(date);
+                    var month = calendar.GetMonth(date);
```

Comment:
Calculate `month` and `year` after determining the `week`. This is too far away from their first use.